lib/smooth/application/display/DisplaySpi.cpp

```diff
@@ -59,87 +58,146 @@ namespace smooth::application::display
             }
-            else
+
+            // check to see if we have to delay before advancing to next sequence in ints_cmds
+            if (data_length >= 0x80)
             {
-                res &= lcd_wr_cmd(ili_init_cmds.at(seq_num).cmd);
-                res &= lcd_wr_data(ili_init_cmds.at(seq_num).data, ili_init_cmds.at(seq_num).databytes & 0x1F);
+                uint8_t delay_data_length = static_cast<uint8_t>(data_length & 0x7F);
+                size_t delay_time = init_cmds[index].data[delay_data_length];
+
+                // check delay_data_length to see if there is data to send before delaying
+                if (delay_data_length != 0)
+                {
+                    res &= send_data(init_cmds[index].data, delay_data_length);
+                }
+
+                // delay value of 255 is really 500
+                delay_time = delay_time == 255 ? 500 : delay_time;
+                std::this_thread::sleep_for(std::chrono::milliseconds(delay_time));
             }
-
-            // advance to next sequence in ili_ints_cmds
-            seq_num++;
         }
 
         return res;
     }
 
-    // Send a command to the LCD. Uses spi_device_polling_write, which waits
-    // until the transfer is complete.
+    // Send muliple commands to the display.
+    // Uses spi_device_polling_write, which waits until the transfer is complete.
+    bool DisplaySpi::send_cmds(const uint8_t* cmds, size_t length)
+    {
+        // reset current transaction counter
+        current_transaction = 0;
+
+        // setup pretrans data/command pin state
+        dc_pretrans_pin_states[0] = PIN_LOW;
+        cs_pretrans_pin_states[0] = PIN_LOW;
+        cs_posttrans_pin_states[0] = PIN_HIGH;
+
+        std::lock_guard<std::mutex> lock(get_guard());
+        spi_transaction_t trans;
+        std::memset(&trans, 0, sizeof(trans));  //Zero out the transaction
+        trans.rx_buffer = nullptr;
+        trans.rxlength = 0;
+        trans.length = length * 8;
+        trans.tx_buffer = cmds;
+
+        return polling_write(trans);
+    }
+
+    // Send a single command to the display.
+    // Uses spi_device_polling_write, which waits until the transfer is complete.
     //
     // Since command transactions are usually small, they are handled in polling
     // mode for higher speed. The overhead of interrupt transactions is more than
     // just waiting for the transaction to complete.
-    bool ILI9341::lcd_wr_cmd(const uint8_t cmd)
+    bool DisplaySpi::send_cmd(const uint8_t cmd)
     {
         // reset current transaction counter
         current_transaction = 0;
 
         // setup pre and post control pin states
-        dc_pretrans_pin_states.at(0) = PIN_LOW;
-        cs_pretrans_pin_states.at(0) = PIN_LOW;
-        cs_posttrans_pin_states.at(0) = PIN_HIGH;
+        dc_pretrans_pin_states[0] = PIN_LOW;
+        cs_pretrans_pin_states[0] = PIN_LOW;
+        cs_posttrans_pin_states[0] = PIN_HIGH;
```

Comment:
These three lines are the same as in `send_cmds` (very similar named btw) and there's a nearly identical set in `send_data` I think it makes sense to break them into a `prepare_pin_command()` and `prepare_pin_data()` or similarly named functions, thoughts?

Reply:
Created two new functions,